pastas/io/base.py

```diff
@@ -150,9 +150,15 @@ def _load_model(data: dict) -> Model:
     # Convert parameters to numeric
     ml.parameters = ml.parameters.infer_objects()
 
-    # When initial values changed
-    for param, value in ml.parameters.loc[:, "initial"].items():
-        ml.set_parameter(name=param, initial=value)
+    # When parameter initial values and bounds changed
+    for pname, data in ml.parameters.iterrows():
```

Comment:
not a big issue, since its the last thing in this function, but you're overwriting data variable from before.

Reply:
good eye